duckduckgo/iOS
# Status view error UI

Task/Issue URL: https://app.asana.com/0/0/1205084446087081/f
Tech Design URL:
CC:

**Description**:

This is the first PR for the Status View error states. The primary purpose of this is to get the design for review ASAP as this is the last task blocking design review. Therefore the main intention is to show _an_ error.

This is initially achieved by just checking the `ConnectionErrorObserver` for a non-nil value, showing the generic connection failed error in that case, then showing no error in the case of a nil value.

A following PR will add handling of connection interruptions, however this will require some BSK changes so I didn’t want to block the design review to wait for that sometimes lengthy process.

**Steps to test this PR**:
1. Ensure you’ve activated NetP by entering an invite code on the Settings -> Network Protection screen.
2. Ensure that Network Protection is switched off
3. Navigate to the Debug menu (Settings -> Debug -> Network Protection) and trigger a tunnel error
4. Go back to the Network Protection Status View and hit connect
5. **Should see an error**
6. Tap connect again
7. **The error should be cleared**

**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


—
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
